deltacat/compute/compactor_v2/steps/merge.py

```diff
@@ -449,25 +456,34 @@ def _materialize(
 
 @ray.remote
 def merge(input: MergeInput) -> MergeResult:
+    with NodeUtilizationOverTimeRange() as node_util:
+        logger.info(f"Starting merge task...")
+
+        # Log node peak memory utilization every 10 seconds
+        def log_peak_memory():
+            logger.debug(
+                f"Node peak memory utilization so far: {node_util.max_memory} ({node_util.max_memory/BYTES_PER_GIBIBYTE} GB)"
+            )
 
-    logger.info(f"Starting merge task...")
-    merge_result, duration = timed_invocation(func=_timed_merge, input=input)
+        node_util.schedule_callback(log_peak_memory, 10)
```

Comment:
can we pass log_peak_memory to constructor of `NodeUtilizationOverTimeRange`?

Reply:
The callback needs to access the member variable of `NodeUtilizationOverTimeRange`, so I figured the current approach would be fine. I think this approach gives a bit more flexibility if you're not immediately ready to schedule a callback.